src/scripts/clipperUI/componentBase.ts

```diff
@@ -89,25 +89,15 @@ export abstract class ComponentBase<TState, TProps> {
 			},
 			onkeyup: (e: KeyboardEvent) => {
 				let element = e.currentTarget as HTMLElement;
-				if (e.which === Constants.KeyCodes.enter) {
-					// Hitting Enter on <a> tags that contains an href automatically fire the click event, so don't do it again
-					if (!(element.tagName === "A" && element.hasAttribute("href"))) {
-						// Intentionally sending click event before handling the method
-						// TODO replace this comment with a test that validates the call order is correct
-						let id = element.id;
-
-						Clipper.logger.logClickEvent(id);
-
-						if (handleMethod) {
-							handleMethod(e);
-						}
-					}
+				if (e.which === Constants.KeyCodes.space) {
+					ComponentBase.triggerActionEvent(element, handleMethod, e);
 				} else if (e.which === Constants.KeyCodes.tab) {
 					// Since they are using the keyboard, revert to the default value of the outline so it is visible
 					element.style.outlineStyle = "";
 				}
 
-				if (!setNameForArrowKeyNav) {
+				if (!setNameForArrowKeyNav && !defaultAction) {
```

Comment:
The existance of setNameForArrowKeyNav and defaultAction seem to be tightly coupled in this method. Do they really need to be?